lock-branch/action.yaml

```diff
@@ -26,36 +26,52 @@ branding:
 runs:
   using: "composite"
   steps:
-  - name: Set up Python
-    uses: actions/setup-python@v4
-    with:
-      python-version: 3.9
-  - name: Install pontos
-    run: |
-      apt-get update && apt-get --assume-yes install python3-venv wget
-      python3 -m venv .venv
-      . .venv/bin/activate
-      python -m pip install --upgrade pip
-      python -m pip install --upgrade pontos
-      wget https://github.com/greenbone/pontos/raw/main/scripts/github/lock-branch.py -O ~/lock-branch.py
-    shell: bash
-  - name: lock/unlock?
-    run: |
-      if [[ "${{inputs.lock}}" == "true" ]];
-      then
-        echo "LOCK_OPT=--lock" >> $GITHUB_ENV
-      elif [[ "${{inputs.lock}}" == "false" ]];
-      then
-        echo "LOCK_OPT=--no-lock" >> $GITHUB_ENV
-      else
-        echo "LOCK_OPT=--no-lock" >> $GITHUB_ENV
-      fi
-    shell: bash
-    # in case of blubber input let the branch be unlocked!
-  - name: Prepare release and store the version
-    run: |
-      . .venv/bin/activate
-      pontos-github-script ~/lock-branch.py ${{ inputs.repository }} ${{ inputs.branch }} ${{ env.LOCK_OPT }}
-    shell: bash
-    env:
-      GITHUB_TOKEN: ${{ inputs.github-token }}
+    - name: Set up Python
+      uses: actions/setup-python@v4
+      id: python
+      with:
+        python-version: "3.10"
+    - name: Virtual Environment
+      id: virtualenv
+      run: |
+        echo "path=${{ github.action_path }}/${{ github.action }}-venv" >> $GITHUB_OUTPUT
+        echo "name=${{ github.action }}-venv" >> $GITHUB_OUTPUT
+      shell: bash
+    - name: Cache Virtual Environment
+      id: cache-virtualenv
+      uses: actions/cache@v3
+      with:
+        key: ${{ runner.os }}-${{ steps.python.outputs.python-version }}-${{ steps.virtualenv.outputs.name }}
+        path: ${{ steps.virtualenv.outputs.path }}
+    - name: Create virtual environment
+      if: ${{ steps.cache-virtualenv.outputs.cache-hit != 'true' }}
+      run: |
+        python -m venv ${{ steps.virtualenv.outputs.path }}
+      shell: bash
```

Comment:
Now this is copy-pasta. Any chance to capsulate this in another action?
`greenbone/actions/python-caching` or sth?

Reply:
I am really not sure if that will work because of the usage of `${{ github.action }}`